frontend/src/components/ErrorCard/ErrorCard.js

```diff
@@ -0,0 +1,41 @@
+import React, { useContext } from 'react';
+
+import { Text, View } from '@instructure/ui';
+
+import { AppContext } from '../AppState';
+
+const ErrorCard = () => {
+  const { state } = useContext(AppContext);
+
+  let message;
+
+  switch (state.processing.errorMessage) {
+    case 'Bad Request':
```

Comment:
Are there examples of dispatched `businessDataFetchErrored` events with this value (and `'No Content'`)? I wasn't quite sure where these were coming from.

Reply:
Right, so there are three main cases coming from our backend that I'm handling here. One thing to note is that I am passing the `status_text` as the `state.processing.errorMessage` in all error cases:
1. The call to the Canvas API fails. In that case I return `HttpResponseBadRequest` which has `status_text` of "Bad Request". So I catch "Bad Request" and return a more verbose message to the user
2. The processing of the data from Canvas fails. In that case I return `HttpResponseNoContent` which has `status_text` of "No Content". Its actually a 204, which is not an error code, but it seemed the most appropriate in this case. Then I follow the same path as #1
3. The last case is an unhandled error, and that is the default message

Reply:
I see -- so the backend Django app is calling Canvas, and then setting up these particular responses to the client-side app. Thanks!